src/main/kotlin/primitives/Node.kt

```diff
@@ -0,0 +1,9 @@
+package primitives
+
+abstract class Node<K : Comparable<K>, V, N : Node<K, V, N>>(k: K, v:V) {
+    var key: K = k
+    var value: V = v
```

Comment:
возможно, стоит занести свойства key, value в конструктор и не использовать лишние переменные

```suggestion
abstract class Node<K : Comparable<K>, V, N : Node<K, V, N>>(
        var key: K,
        var value: V,
) {
```